src/base/static/components/organisms/activity-stream.js

```diff
@@ -86,74 +56,48 @@ class ActivityStream extends Component {
     );
   }
 
-  onAddAction({ model, slug, collectionId }) {
-    // Get the id of the place related to this action.
-    model.attributes[this.placeIdProperty] = model.get(
-      constants.TARGET_PROPERTY_NAME,
-    ).place
-      ? model
-          .get(constants.TARGET_PROPERTY_NAME)
-          .place.split("/")
-          .slice(-1)[0]
-      : model.get(constants.TARGET_PROPERTY_NAME).id;
-    model.attributes[this.slugProperty] = slug;
-    model.attributes[this.collectionIdProperty] = collectionId;
-
-    // When we get a new activity model, just unshift it to the front of the
-    // List of serialized models.
-    this.setState({
-      activityModels: this.state.activityModels.unshift(
-        fromJS(model.attributes),
-      ),
-    });
-  }
-
   render() {
     return (
       <ul style={{ paddingLeft: 0, marginTop: 0 }}>
-        {this.state.activityModels.map((activityModel, i) => {
-          const submitterName = activityModel
-            .get(constants.TARGET_PROPERTY_NAME)
-            .get(constants.SUBMITTER_NAME);
-          const target = activityModel.get(constants.TARGET_PROPERTY_NAME);
-          const collectionId = activityModel.get(this.collectionIdProperty);
-          const placeId = activityModel.get(this.placeIdProperty);
-          const slug = activityModel.get(this.slugProperty);
+        {this.props.activity.map((activity, i) => {
+          const target = activity.target;
```

Comment:
FYI I started phasing out the usage of the `constants` export, as I don't think it's really that helpful, and makes the code harder to read.